### DISCUSSION TITLE
My channel improvements

### DISCUSSION THREAD

**Discussion by RoEdAl** [IchthysMaranatha/asterisk-chan-quectel#46] [General]
[Here](//github.com/RoEdAl/asterisk-chan-quectel) you can find my fork of this project with many changes (improvements?).

Supported modules (boards):

* *Quectel* EC25-E,
* *SimCOM* SIM7600G-H,
* *SimCOM* SIM900.

Most important changes are noted in [README](//github.com/RoEdAl/asterisk-chan-quectel/blob/quectel-response/README.md).

**Comment by RSUN** [comment]
Hello @RoEdAl , I would like to try your repo improvements
Could you please expand on the cmake installation commands? In particular, how do I specify a particular folder path to the asterisk compiled source. 

For example, this repo has the following instructions

> If you did not make install Asterisk in the usual location and configure cannot find the asterisk header files in /usr/include/asterisk, you may optionally pass --with-asterisk=PATH/TO/INCLUDE.

How would you do that with cmake?

**Reply by RoEdAl** [reply]
Currently there's no option to specify alternate location of *Asterisk* headers.
You have to edit `CMakeLists.txt` manually and add [`include_directories`](https://cmake.org/cmake/help/latest/command/include_directories.html) command:

```cmake
#
# asterisk-chan-quectel
#

CMAKE_MINIMUM_REQUIRED(VERSION 3.14)
include_directories(SYSTEM /path/to/asterisk-h)
```

Do you really need to specify nonstandard location of header files? Just install Asterisk headers via `sudo make install-headers` command. On *Ubuntu* you can just install `asterisk-dev` package. See also my [install-asterisk-headers](https://github.com/RoEdAl/asterisk-chan-quectel/blob/master/.github/actions/install-asterisk-headers/action.yml) action (currently unused) which demonstrates how to download, extract, configure, build and install header files.
